src/dataServices/hybridDataService.ts

```diff
@@ -0,0 +1,323 @@
+                separatedResourceData.bulkDataLink,
+                'application/json',
+            );
+            const createResponse = await this.dbPersistenceService.createResourceWithIdNoClone(
+                resourceType,
+                separatedResourceData.resource,
+                resourceId,
+                tenantId,
+            );
+            // Update the meta data
+            resourceClone.meta = createResponse.resource.meta;
+            return {
+                success: createResponse.success,
+                message: createResponse.message,
+                resource: resourceClone,
+            };
+        } catch (e) {
+            console.log(`Creation of ${resource.resourceType}: ${resource.id} has failed`);
+            await this.dbPersistenceService.deleteResource({
+                resourceType,
+                id: separatedResourceData.resource.id,
+            });
+            throw e;
+        }
+    }
+
+    async updateResource(request: UpdateResourceRequest) {
+        if (!this.shallStoreOnObjectStorage(request.resourceType)) {
+            return this.dbPersistenceService.updateResource(request);
+        }
+        this.assertValidTenancyMode(request.tenantId);
+        const { resourceType, resource, id, tenantId } = request;
+        try {
+            // Will throw ResourceNotFoundError if resource can't be found
+            await this.dbPersistenceService.readResource({ resourceType, id, tenantId });
+        } catch (e) {
+            if (this.updateCreateSupported && isResourceNotFoundError(e)) {
+                return this.createResourceWithId(resourceType, resource, id, tenantId);
+            }
+            throw e;
+        }
+        const resourceClone = clone(resource);
+        // Separate the main payload from the resource.
+        const separatedResourceData = this.separatePayloadFromResource(resourceType, resourceClone, id, tenantId);
+        try {
+            // Ensure the order: first S3 then ddb to avoid possible data races
+            await S3ObjectStorageService.uploadObject(
+                separatedResourceData.bulkData,
+                separatedResourceData.bulkDataLink,
+                'application/json',
+            );
+            const updateResponse = await this.dbPersistenceService.updateResourceNoCheckForExistenceNoClone(
+                resourceType,
+                separatedResourceData.resource,
+                id,
+                tenantId,
+            );
+            // Update the meta data
+            resourceClone.meta = updateResponse.resource.meta;
+
+            return {
+                success: updateResponse.success,
+                message: updateResponse.message,
+                resource: resourceClone,
+            };
+        } catch (e) {
+            console.log(`Update of ${resource.resourceType}: ${resource.id} has failed`);
+            await this.dbPersistenceService.deleteResource({ resourceType: request.resourceType, id });
```

Comment:
Why delete here if update fails? This would mean the resource completely disappears if update fails, or?